modules/simple_forms_api/app/services/simple_forms_api/s3_service/archive_submission_to_pdf.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+# To use
+# ids = <array of submission ids to archive>
+# parent_dir = <the name of the s3 'folder' where these archives will be put>
+#
+# to see your archive in s3
+# 1. go here https://console.amazonaws-us-gov.com/s3/home?region=us-gov-west-1#
+# 2. login with 2fa
+# 3. search for dsva-vetsgov-prod-reports
+# 4. search for your parent_dir name, e.g. 526archive_aug_21st_2024
+#
+# If you do not provide a parent_dir, the script defaults to a folder called vff-simple-forms
+#
+# OPTION 1: Run the script with user groupings
+# - requires SubmissionDuplicateReport object
+# - SubmissionArchiveHandler.new(submission_ids: ids, parent_dir:).run
+#
+# OPTION 2: Run without user groupings
+# ids.each { |id| ArchiveSubmissionToPdf.new(submission_id: id, parent_dir:).run }
+# this will just put each submission in a folder by it's id under the parent dir
+module SimpleFormsApi
+  module S3Service
+    class ArchiveSubmissionToPdf < Utils
+      attr_reader :failures, :include_json_archive, :include_text_archive, :metadata,
+                  :parent_dir, :quiet_pdf_failures, :quiet_upload_failures, :run_quiet,
+                  :submission
+
+      VALID_VFF_FORMS = %w[
+        20-10206 20-10207 21-0845 21-0966 21-0972 21-10210
+        21-4138 21-4142 21P-0847 26-4555 40-0247 40-10007
+      ].freeze
+
+      def initialize(submission_id: nil, submission: nil, **options)
+        defaults = default_options.merge(options)
+
+        @failures = []
+        @submission = submission || FormSubmission.find(submission_id)
+
+        assign_instance_variables(defaults)
+      end
+
+      def run
+        log_info("Processing submission ID: #{submission.id}")
+        process_submission_files
+        output_directory_path
+      rescue => e
+        handle_error("Failed submission: #{submission.id}", e, { submission_id: submission.id })
+      end
+
+      private
+
+      def default_options
+        {
+          file_path: nil, # file path for the PDF file to be archived
+          include_json_archive: true, # include the form data as a JSON object
+          include_text_archive: true, # include the form data as a text file
+          metadata: {},
+          parent_dir: 'vff-simple-forms',
+          quiet_pdf_failures: true, # skip PDF generation silently
+          quiet_upload_failures: true, # skip problematic uploads silently
+          run_quiet: true, # silence but record errors, logged at the end
```

Comment:
With all the stuff over the last year with the code yellow and silent failures, I think VA people blow a gasket if they saw these lol, especially defaulted to true.

Reply:
Do you think it'd be better just to get rid of the option to silence failures or just change the defaults? I could also add a "dry_run" option if that seems more appropriate.

Reply:
Im not a huge fan of hard-coding stuff to support testing into where it could potentially get used in prod, but at the very least would default it false so that if anyone uses this without knowing about it, they are not by default in a state of silently failing.

If I needed to do things to test I would mock of the calls in the rspec, or just let it do the thing. There is mocked S3 returns in the mockdata repo that allow it to be coded like it was actually uploading to S3, and the testing framework/mock-data repo just intercepts the request and knows to return a thing that looks like an S3 thing. 

For example: 
https://github.com/department-of-veterans-affairs/vets-api/blob/02ecc749fcea88c9527114fd85ba92d31bb6db1f/spec/uploaders/form1010cg/poa_uploader_spec.rb#L125